src/solver/optimisation/opt_appel_solveur_lineaire.cpp

```diff
@@ -277,14 +270,49 @@ bool OPT_AppelDuSimplexe(const OptimizationOptions& options,
             ProbSpx = nullptr;
             solver = nullptr;
             PremierPassage = false;
-            goto RESOLUTION;
+            return false;
         }
 
         else
         {
             throw FatalError("Internal error: insufficient memory");
         }
     }
+    return true;
+}
+
+bool OPT_AppelDuSimplexe(const OptimizationOptions& options,
+                         PROBLEME_HEBDO* problemeHebdo,
+                         int NumIntervalle,
+                         const int optimizationNumber,
+                         const OptPeriodStringGenerator& optPeriodStringGenerator,
+                         IResultWriter& writer)
+{
+    PROBLEME_ANTARES_A_RESOUDRE* ProblemeAResoudre = problemeHebdo->ProblemeAResoudre;
+    Optimization::PROBLEME_SIMPLEXE_NOMME Probleme(ProblemeAResoudre->NomDesVariables,
+                                                   ProblemeAResoudre->NomDesContraintes,
+                                                   ProblemeAResoudre->StatutDesVariables,
+                                                   ProblemeAResoudre->StatutDesContraintes);
+    bool PremierPassage = true;
```

Comment:
This was only useful when using a `goto`. I strongly suggest to change `bool&` to `bool` in `OPT_TryToCallSimplex`'s signature. Then pass `true` in the first call, `false` in the 2nd one.